squirrels/_auth.py

```
@@ -468,6 +411,298 @@ def get_user_from_token(self, token: str | None) -> tuple[RegisteredUser | None,
+    def _get_provider_metadata(self, provider_name: str) -> dict:
+        provider = next((p for p in self.auth_providers if p.name == provider_name), None)
+        if provider is None:
+            raise InvalidInputError(404, "auth_provider_not_found", f"Provider '{provider_name}' not found")
+
+        metadata_url = provider.provider_configs.server_metadata_url
+        cached = self._provider_metadata_cache.get(metadata_url)
+        if isinstance(cached, dict) and cached:
+            return cached
+
+        try:
+            response = requests.get(metadata_url)
```

Ruff (0.14.13)

Probable use of `requests` call without timeout

(S113)